### ISSUE TITLE
Replace custom `xcconfig` parsing in favor of using `xcodeproj` gem

### ISSUE THREAD

**Issue by mokagio** [wordpress-mobile/release-toolkit#470]
Extracted from https://github.com/wordpress-mobile/release-toolkit/pull/458#discussion_r1178568904:

Our current logic to extract version values from `xcconfig` files relies on a custom parsing of the config file:

https://github.com/wordpress-mobile/release-toolkit/blob/e4b28a48dc760fe25be1d31d959de8c0b1726821/lib/fastlane/plugin/wpmreleasetoolkit/helper/ios/ios_version_helper.rb#L286-L302

The implementation is too rigid, requiring the `xcconfig` files to be in the `key=value` format, even though `key = value` is a valid _and more readable_ syntax.

We should be able to replace that bespoke logic with the API provided by the `xcodeproj` gem like [we do in Jetpack and WordPress iOS](https://github.com/wordpress-mobile/WordPress-iOS/blob/00a91b905cd1614cb272224114464fda6292aa77/fastlane/lanes/build.rb#L320-L321):

```ruby
version_config_path = File.join(PROJECT_ROOT_FOLDER, 'config', 'Version.internal.xcconfig')
versions = Xcodeproj::Config.new(File.new(version_config_path)).to_hash
```

`xcodeproj` is already available to us as it is a Fastlane dependency.

**Comment by spencertransier** [comment]
It looks like @iangmaia has already started work on this here: https://github.com/wordpress-mobile/release-toolkit/pull/451

**Comment by mokagio** [comment]
Great find @spencertransier !

I guess there is some code from @iangmaia's trial that is yet to be merged. This is not urgent as far as I'm concerned so I might leave it up to Ian to follow up after his onboarding.

**Comment by spencertransier** [comment]
This was implemented in the `ios_version_file` class in #512

**Comment by mokagio** [comment]
🎉
